src/dmd/target.d

```diff
@@ -216,7 +237,48 @@ extern (C++) struct Target
         else
             assert(0, "unknown environment");
     }
-
+    /**
+     * Determine the instruction set to be used
+     */
+    void setCPU()
+    {
+        if(!isXmmSupported())
+        {
+            cpu = CPU.x87;   // cannot support other instruction sets
+            return;
+        }
+        switch (cpu)
+        {
+            case CPU.baseline:
+                cpu = CPU.sse2;
+                break;
+            
+            case CPU.native:
+            {
+                import core.cpuid;
+                cpu = core.cpuid.avx2 ? CPU.avx2 :
+                      core.cpuid.avx  ? CPU.avx  :
+                                        CPU.sse2;
+                break;
+            }
+            default:
+                break;
+        }
+    }
+    /**
+     * Add predefined global identifiers that are determied by the target
+     */
+    void addPredefinedGlobalIdentifiers()
```

Comment:
Including OS and runtime versions in the future? :-)

Reply:
ofc